gobblin-core/src/main/java/gobblin/writer/FsDataWriter.java

```diff
@@ -49,17 +59,42 @@ public FsDataWriter(State properties, String fileName, int numBranches, int bran
     // Add all job configuration properties so they are picked up by Hadoop
     JobConfigurationUtils.putStateIntoConfiguration(properties, conf);
 
-    // Initialize file system
-    String uri = properties.getProp(
-        ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, numBranches, branchId),
-        ConfigurationKeys.LOCAL_FS_URI);
-    this.fs = FileSystem.get(URI.create(uri), conf);
+    String uri =
+        properties
+            .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, numBranches,
+                branchId), ConfigurationKeys.LOCAL_FS_URI);
+
+    if (properties.getPropAsBoolean(ConfigurationKeys.FS_AS_PROXY_USER, ConfigurationKeys.DEFAULT_FS_AS_PROXY_USER)) {
+      // Initialize file system as a proxy user. Both properties PROXY_USER_NAME and TOKEN_FILE_NAME should be set in Source.
+      Preconditions.checkArgument(StringUtils.isNotBlank(properties.getProp(ConfigurationKeys.PROXY_USER_NAME)),
+          "State does not contain a proper proxy user name");
+      Preconditions.checkArgument(StringUtils.isNotBlank(properties.getProp(ConfigurationKeys.TOKEN_FILE_NAME)),
```

Comment:
Can we move these precondition checks into `ProxiedFileSystemWrapper`, that way we don't have to re-implement these checks each place `ProxiedFileSystemWrapper` is used